pkg/controller/controller.go

```diff
@@ -121,22 +128,27 @@ func (dc *DefaultController) processItem(e entitystore.Entity) error {
 	if !ok {
 		return errors.Errorf("trying to process an entity with no entity handler: %v", reflect.TypeOf(e))
 	}
+	if e.GetDelete() {
```

Comment:
I think the Delete is a legacy flag that should be cleaned, it should be replaced by the status ``StatusDELETING ``

Reply:
I'm not sure about this.  I don't really like overloading the status field with the "requested state" which is what we are doing here.  The current state when deleting something is still "READY", but we are flagging it for deletion.  This is going to be a bigger conversation at some point, where we may need to more formally use requested state (currently the spec field on the entity) to reflect the desired state, while status is simply the current state.

Reply:
I tend to agree with @berndtj on this.